logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/call/usecase/GetIncomingCallsUseCase.kt

```diff
@@ -1,21 +1,59 @@
 package com.wire.kalium.logic.feature.call.usecase
 
 import com.wire.kalium.logic.data.call.CallRepository
-import com.wire.kalium.logic.feature.call.Call
+import com.wire.kalium.logic.data.conversation.ConversationDetails
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.feature.call.CallStatus
+import com.wire.kalium.logic.functional.flatMapFromIterable
 import com.wire.kalium.logic.sync.SyncManager
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
 
 interface GetIncomingCallsUseCase {
-    suspend operator fun invoke(): Flow<List<Call>>
+    suspend operator fun invoke(): Flow<List<IncomingCallData>>
 }
 
 internal class GetIncomingCallsUseCaseImpl(
+    private val conversationRepository: ConversationRepository,
     private val callRepository: CallRepository,
     private val syncManager: SyncManager
 ) : GetIncomingCallsUseCase {
 
-    override suspend operator fun invoke(): Flow<List<Call>> {
+    //TODO update UnitTests after fixing all the other TODOs
+    override suspend operator fun invoke(): Flow<List<IncomingCallData>> {
         syncManager.waitForSlowSyncToComplete()
         return callRepository.getIncomingCalls()
+            .flatMapLatest {
+                it.flatMapFromIterable { call ->
+                    conversationRepository.getConversationDetailsById(call.conversationId)
+                        .map { conversationsDetails ->
+                            val callerName: String
+                            val callerTeamName: String?
+                            when (conversationsDetails) {
+                                is ConversationDetails.OneOne -> {
+                                    callerTeamName = "" //TODO get team name by the teamId
+                                    callerName = conversationsDetails.otherUser.name ?: "Someone"
+                                }
+                                else -> {
+                                    callerTeamName = "" //TODO get team name by the teamId
+                                    callerName = "Someone" //TODO get caller name for the group conversations
+                                }
+                            }
+                            IncomingCallData(call.conversationId, call.status, conversationsDetails, callerName, callerTeamName)
+                        }
+                }
+            }
+            .distinctUntilChanged()
     }
 }
+
+data class IncomingCallData(
+    val conversationId: ConversationId,
+    val status: CallStatus,
+    val conversationDetails: ConversationDetails,
+    val callerName: String,
+    val callerTeamName: String?
+)
```

Comment:
Then we will need to create a separat data class for `outgoingCall` and `initiatedCall` ?

I would suggest to do it in different way, instead of fetching call details every time we invoke the incomingCallUseCase or whatever, we can can do that while creating the call object. This way Call object will hold of informations needed to be displayed

In the **callManagerRefactor** PR, we extracted that part of creating the call so you can do it there.